src/main/java/com/netflix/frigga/Names.java

```diff
@@ -31,6 +31,31 @@
     private static final Pattern NAME_PATTERN = Pattern.compile(
             "^([" + NameConstants.NAME_CHARS + "]+)(?:-([" + NameConstants.NAME_CHARS + "]*)(?:-(["
                     + NameConstants.NAME_HYPHEN_CHARS + "]*?))?)?$");
+    private static final Pattern LABELED_COUNTRIES_PATTERN = Pattern.compile(
```

Comment:
feels like these could benefit from a `static Pattern createLabeledVariablePattern(String label) {` that handled all the duplication in this pattern creation

Reply:
Thanks @cfieber, makes sense, updated!